src/Helpers/EventHelper.ts

```diff
@@ -0,0 +1,159 @@
+import EventApi from "Api/Event.Api";
+import { C_EVENT_CREEP_SPAWNED, C_EVENT_BUILD_COMPLETE, STANDARD_SQUAD, STALKER_SOLO } from "utils/Constants";
+import { STANDARD_SQUAD_ARRAY, ZEALOT_SOLO_ARRAY, STALKER_SOLO_ARRAY } from "utils/militaryConfig";
+import MemoryApi from "Api/Memory.Api";
+import MemoryHelper_Room from "./MemoryHelper_Room";
+import UserException from "utils/UserException";
+
+export default class EventHelper {
+
+    /**
+     * process event triggered for ramparts built in a room
+     * @param room the room the event occured in
+     * @param event the event that occured
+     * @param constructionSite the site we are triggering the event for
+     */
+    public static rampartBuiltTrigger(room: Room, event: CustomEvent, siteBuilt: Structure): void {
+
+        // Update upgrade jobs and fire all the creeps
+        MemoryHelper_Room.updateWorkPart_buildJobs(room);
+        MemoryApi.cleanCreepDeadJobsMemory(room.name);
+    }
+
+    /**
+     * military creep successfully spawned event
+     * this function is a straight mess, will need refactoring once bugs are ironed out
+     * @param room the room the event occured in
+     * @param event the event that occured
+     * @param creep the creep thaat just spawned
+     */
+    public static miltaryCreepSpawnTrigger(room: Room, event: CustomEvent, creep: Creep): void {
+
+        // Early return if creep has no options for some reason
+        if (!creep.memory.options) {
+            return;
+        }
+
+        // Get all attack flag memory associated with the room (should only be 1, but plan for multiple possible in future)
+        const attackRoomFlags: AttackFlagMemory[] = MemoryApi.getAllAttackFlagMemoryForHost(room.name);
+        const creepOptions: CreepOptionsMili = creep.memory.options! as CreepOptionsMili;
+        const creepRole: RoleConstant = creep.memory.role;
+        const creepIsSquadMember: boolean = creepOptions.squadSize !== 0;
+        let requestingFlag: AttackFlagMemory | undefined;
+
+        // Find the one that requested this creep and handle it accordingly
+        // !TODO Before Merging pullrequest and finish bug fixes
+        // Move this into a function, no reason the function should be this crowded with nonsense
+        // Also, brak plz, help me find a way to standarize this so we don't have to edit some cryptic function
+        // every time we want to add a military role
+        // If not removing that entirely, at least help make it easy to edit please lol
+        for (const attackFlag of attackRoomFlags) {
+
```

Comment:
Need to add a case here to make sure the creeps target room is the same as the attack flags room name, so we don't mix up squads